hy/compiler.py

```diff
@@ -1942,42 +1934,65 @@ def compile_maths_expression(self, expression):
         if PY35:
             ops.update({"@": ast.MatMult})
 
-        inv = expression.pop(0)
-        op = ops[inv]
+        op = ops[expression.pop(0)]
+        right_associative = op == ast.Pow
 
+        if right_associative:
+            expression = list(expression[::-1])
```

Comment:
Why is the outer `list` here necessary? `[::-1]` should already be returning a list anyway.